### PR TITLE
Merge latest changes from upstream repo

### LINKED ISSUE #6: Merge latest changes from upstream repo
### Description
As the upstream repository has released a new tag [v0.9.0](https://github.com/base-org/node/releases/tag/v0.9.0), all the necessary changes need to be merged to keep the repository up-to-date.

### Acceptance Criteria

- All required changes should merge

### Additional Information
https://github.com/base-org/node/compare/v0.8.4...v0.9.0

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?

This PR resolves #6 

### How was it solved?

- [x] Merge latest changes from upstream repo
  - [x] Add support to run node with `reth` client 
  - [x] Update readme 
  - [x] Update dockerfile
  - [x] Update entrypoint file for `geth` and `reth`

### How was it tested?
Local + Docker

### CODE REVIEW

File: reth/reth-entrypoint [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+AUTHRPC_PORT="${AUTHRPC_PORT:-8551}"
+METRICS_PORT="${METRICS_PORT:-6060}"
+
+if [[ -z "$RETH_CHAIN" ]]; then
+    echo "expected RETH_CHAIN to be set" 1>&2
+    exit 1
+fi
+
+if [ "${OP_NODE_OVERRIDE_CANYON+x}" = x ]; then
+    ADDITIONAL_ARGS="$ADDITIONAL_ARGS --override.canyon=$OP_NODE_OVERRIDE_CANYON"
```
Comment (sameersubudhi) [inline reth/reth-entrypoint:16]: I don't see the `ADDITIONAL_ARGS` getting used anywhere.

File: reth/reth-entrypoint [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  --http.port="$RPC_PORT" \
+  --http.api=debug,eth,net,txpool \
+  --authrpc.addr=0.0.0.0 \
+  --authrpc.port="$AUTHRPC_PORT" \
+  --authrpc.jwtsecret="$OP_NODE_L2_ENGINE_AUTH" \
+  --metrics=0.0.0.0:"$METRICS_PORT" \
+  --chain="$GENESIS_FILE_PATH" \
+  --disable-discovery \
+  --rollup.sequencer-http=$SEQUENCER_HTTP \
+  --rollup.disable-tx-pool-gossip
```
Comment (sameersubudhi) [inline reth/reth-entrypoint]: Probably, something like this?
```suggestion
  --rollup.disable-tx-pool-gossip \
  $ADDITIONAL_ARGS
```

File: reth/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+RUN apt-get update && apt-get -y upgrade && apt-get install -y git libclang-dev pkg-config curl build-essential
+
+ENV REPO=https://github.com/paradigmxyz/reth.git
+ENV VERSION=v1.0.0
+ENV COMMIT=83d412da70af678a46f368533b6df45a287a1ce6
+RUN git clone $REPO --branch $VERSION --single-branch . && \
+    git switch -c branch-$VERSION && \
+    bash -c '[ "$(git rev-parse HEAD)" = "$COMMIT" ]'
+
+RUN cargo build --bin op-reth --locked --features $FEATURES --profile release
```
Comment (sameersubudhi) [inline reth/Dockerfile:30]: Let's create an issue to update the profile to `maxperf` for when it starts working properly again.

File: op-node-entrypoint [hunk 4]
```diff
@@ -22,14 +22,14 @@ get_public_ip() {
...
 }
 
-# if [[ -z "$OP_NODE_NETWORK" ]]; then
+# if [[ -z "$OP_NODE_NETWORK" && -z "$OP_NODE_ROLLUP_CONFIG" ]]; then
 #   echo "expected OP_NODE_NETWORK to be set" 1>&2
 #   exit 1
 # fi
 
-# wait until local geth comes up (authed so will return 401 without token)
+# wait until local execution client comes up (authed so will return 401 without token)
```
Comment (sameersubudhi) [inline op-node-entrypoint]:
```suggestion
# wait until the local execution client comes up (authed so will return 401 without token)
```

File: op-node-entrypoint [hunk 5]
```diff
@@ -22,14 +22,14 @@ get_public_ip() {
...
 until [ "$(curl -s -w '%{http_code}' -o /dev/null "${OP_NODE_L2_ENGINE_RPC/ws/http}")" -eq 401 ]; do
-  echo "waiting for geth to be ready"
+  echo "waiting for execution client to be ready"
```
Comment (sameersubudhi) [inline op-node-entrypoint]:
```suggestion
  echo "waiting for the execution client to be ready"
```

File: README.md [hunk 6]
```diff
@@ -46,7 +46,8 @@ cd lisk-node
 
 1. Ensure you have an Ethereum L1 full node RPC available (not Lisk), and set the `OP_NODE_L1_ETH_RPC` and the `OP_NODE_L1_BEACON` variables (within the `.env.*` files, if using docker-compose). If running your own L1 node, it needs to be synced before the Lisk node will be able to fully sync.
 2. Please ensure that the environment file relevant to your network (`.env.sepolia`, or `.env.mainnet`) is set for the `env_file` properties within `docker-compose.yml`. By default, it is set to `.env.mainnet`.
-3. Run:
+3. Please ensure to set `CLIENT` before starting the node (available options are geth and reth). By default, it is set to `geth`.
```
Comment (sameersubudhi) [inline README.md]:
```suggestion
3. We currently support running either the `op-geth` or the `op-reth` nodes alongside the `op-node`. By default, we run the `op-geth` node. If you would like to run the `op-reth` node instead, please set the `CLIENT` environment variable to `reth` before starting the node. 
```

File: README.md [hunk 7]
```diff
@@ -173,6 +175,68 @@ For, Lisk Mainnet:
...
+  --http.addr=0.0.0.0 \
+  --http.port=8545 \
+  --http.api=debug,eth,net,txpool \
+  --authrpc.addr=0.0.0.0 \
+  --authrpc.port=8551 \
+  --authrpc.jwtsecret=PATH_TO_JWT_TEXT_FILE \
+  --metrics=0.0.0.0:606 \
+  --chain=PATH_TO_NETWORK_GENESIS_FILE \
+  --disable-discovery \
+  --rollup.sequencer-http=https://rpc.sepolia-api.lisk.com \
```
Comment (sameersubudhi) [inline README.md]: We could leave it as a variable but recommend the office sequencer endpoint from Lisk.

File: README.md [hunk 8]
```diff
@@ -173,6 +175,68 @@ For, Lisk Mainnet:
...
+  --http.addr=0.0.0.0 \
+  --http.port=8545 \
+  --http.api=debug,eth,net,txpool \
+  --authrpc.addr=0.0.0.0 \
+  --authrpc.port=8551 \
+  --authrpc.jwtsecret=PATH_TO_JWT_TEXT_FILE \
+  --metrics=0.0.0.0:606 \
+  --chain=PATH_TO_NETWORK_GENESIS_FILE \
+  --disable-discovery \
+  --rollup.sequencer-http=https://rpc.api.lisk.com \
```
Comment (sameersubudhi) [inline README.md]: Same as above.

File: README.md [hunk 9]
```diff
@@ -173,6 +175,68 @@ For, Lisk Mainnet:
...
+  --ws.api=debug,eth,net,txpool \
+  --http \
+  --http.corsdomain="*" \
+  --http.addr=0.0.0.0 \
+  --http.port=8545 \
+  --http.api=debug,eth,net,txpool \
+  --authrpc.addr=0.0.0.0 \
+  --authrpc.port=8551 \
+  --authrpc.jwtsecret=PATH_TO_JWT_TEXT_FILE \
+  --metrics=0.0.0.0:606 \
```
Comment (sameersubudhi) [inline README.md]:
```suggestion
  --metrics=0.0.0.0:6060 \
```

File: README.md [hunk 10]
```diff
@@ -173,6 +175,68 @@ For, Lisk Mainnet:
...
```
Comment (sameersubudhi) [inline README.md]:
```suggestion
  --metrics=0.0.0.0:6060 \
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

File: reth/Dockerfile (hunk 3 above)
Comment (nagdahimanshu) [reply reth/Dockerfile:30]: Created https://github.com/LiskHQ/lisk-node/issues/9

**Review by sameersubudhi — APPROVED** [review]
LGTM 👍

**Review by shuse2 — APPROVED** [review]

**Pull request merged**